crates/shared/src/price_estimation/zeroex.rs

```diff
@@ -40,7 +57,28 @@ impl PriceEstimating for ZeroExPriceEstimator {
         &'a self,
         queries: &'a [Query],
     ) -> futures::stream::BoxStream<'_, (usize, PriceEstimateResult)> {
-        self.0.estimates(queries)
+        if !self.buy_only {
+            self.inner.estimates(queries)
+        } else {
+            async_stream::stream! {
+                let (sell, buy) = queries
```

Comment:
Would it maybe be slightly nicer to drag the `bool` all the way through to the `ZeroexTradeFinder`? Then it could simply return an error for sell quotes on the lowest level and you wouldn't have to implement this additional stream logic.